dbos/_core/workflow.py

```diff
@@ -0,0 +1,488 @@
+            except Exception:
+                dbos.logger.error(
+                    f"Exception encountered in asynchronous workflow: {traceback.format_exc()}"
+                )
+                raise
+
+
+def execute_workflow_id(dbos: "DBOS", workflow_id: str) -> "WorkflowHandle[Any]":
+    status = dbos._sys_db.get_workflow_status_sync(workflow_id)
+    if not status:
+        raise DBOSRecoveryError(workflow_id, "Workflow status not found")
+    inputs = dbos._sys_db.get_workflow_inputs_sync(workflow_id)
+    if not inputs:
+        raise DBOSRecoveryError(workflow_id, "Workflow inputs not found")
+    wf_func = dbos._registry.workflow_info_map.get(status["name"], None)
+    if not wf_func:
+        raise DBOSWorkflowFunctionNotFoundError(
+            workflow_id, "Workflow function not found"
+        )
+    with DBOSContextEnsure():
+        ctx = assert_current_dbos_context()
+        request = status["request"]
+        ctx.request = (
+            serialization.deserialize(request) if request is not None else None
+        )
+        if status["config_name"] is not None:
+            config_name = status["config_name"]
+            class_name = status["class_name"]
+            iname = f"{class_name}/{config_name}"
+            if iname not in dbos._registry.instance_info_map:
+                raise DBOSWorkflowFunctionNotFoundError(
+                    workflow_id,
+                    f"Cannot execute workflow because instance '{iname}' is not registered",
+                )
+            with SetWorkflowID(workflow_id):
+                return start_workflow(
+                    dbos,
+                    wf_func,
+                    status["queue_name"],
+                    True,
+                    dbos._registry.instance_info_map[iname],
+                    *inputs["args"],
+                    **inputs["kwargs"],
+                )
+        elif status["class_name"] is not None:
+            class_name = status["class_name"]
+            if class_name not in dbos._registry.class_info_map:
+                raise DBOSWorkflowFunctionNotFoundError(
+                    workflow_id,
+                    f"Cannot execute workflow because class '{class_name}' is not registered",
+                )
+            with SetWorkflowID(workflow_id):
+                return start_workflow(
+                    dbos,
+                    wf_func,
+                    status["queue_name"],
+                    True,
+                    dbos._registry.class_info_map[class_name],
+                    *inputs["args"],
+                    **inputs["kwargs"],
+                )
+        else:
+            with SetWorkflowID(workflow_id):
+                return start_workflow(
+                    dbos,
+                    wf_func,
+                    status["queue_name"],
+                    True,
+                    *inputs["args"],
+                    **inputs["kwargs"],
+                )
+
+
+def start_workflow(
```

Comment:
Is  there a start_workflow_async that I missed?

Reply:
There is not. `start_workflow` is already async-ish in that it returns a `WorkflowHandle`. Seemed odd to have a version that returned an `Awaitable[WorkflowHandle]`

Reply:
Yes, but that is what is necessary to allow the event loop to proceed while the database call to record the workflow status is occurring.

Reply:
Yeah, it's 100% necessary